src/bezier.jl

```diff
@@ -0,0 +1,691 @@
+rotate(l::LineTo, a) = LineTo(rotmatrix2d(a) * l.p)
+function rotate(c::CurveTo, a)
+    m = rotmatrix2d(a)
+    CurveTo(m * c.c1, m * c.c2, m *c.p)
+end
+function rotate(e::EllipticalArc, a)
+    m = rotmatrix2d(a)
+    newc = m * e.c
+    newangle = e.angle + a
+    EllipticalArc(newc, e.r1, e.r2, newangle, e.a1, e.a2)
+end
+rotate(b::BezierPath, a) = BezierPath(PathCommand[rotate(c::PathCommand, a) for c in b.commands])
+
+function fit_to_bbox(b::BezierPath, bb_target::Rect2; keep_aspect = true)
+    bb_path = bbox(b)
+    ws_path = widths(bb_path)
+    ws_target = widths(bb_target)
+
+    center_target = origin(bb_target) + 0.5 * widths(bb_target)
+    center_path = origin(bb_path) + 0.5 * widths(bb_path)
+
+    scale_factor = ws_target ./ ws_path
+    scale_factor_aspect = if keep_aspect
+        min.(scale_factor, minimum(scale_factor))
+    else
+        scale_factor
+    end
+
+    bb_t = translate(scale(translate(b, -center_path), scale_factor_aspect), center_target)
+end
+
+function fit_to_unit_square(b::BezierPath, keep_aspect = true)
+    fit_to_bbox(b, Rect2((0.0, 0.0), (1.0, 1.0)), keep_aspect = keep_aspect)
+end
+
+Base.:+(pc::EllipticalArc, p::Point2) = EllipticalArc(pc.c + p, pc.r1, pc.r2, pc.angle, pc.a1, pc.a2)
+Base.:+(pc::ClosePath, p::Point2) = pc
+Base.:+(bp::BezierPath, p::Point2) = BezierPath(bp.commands .+ Ref(p))
+
+# markers that fit into a square with sidelength 1 centered on (0, 0)
+
+const BezierCircle = let
+    r = 0.47 # sqrt(1/pi)
+    BezierPath([
+        MoveTo(Point(r, 0.0)),
+        EllipticalArc(Point(0.0, 0), r, r, 0.0, 0.0, 2pi),
+        ClosePath(),
+    ])
+end
+
+const BezierUTriangle = let
+    aspect = 1
+    h = 0.97 # sqrt(aspect) * sqrt(2)
+    w = 0.97 # 1/sqrt(aspect) * sqrt(2)
+    # r = Float32(sqrt(1 / (3 * sqrt(3) / 4)))
+    p1 = Point(0, h/2)
+    p2 = Point2(-w/2, -h/2)
+    p3 = Point2(w/2, -h/2)
+    centroid = (p1 + p2 + p3) / 3
+    bp = BezierPath([
+        MoveTo(p1 - centroid),
+        LineTo(p2 - centroid),
+        LineTo(p3 - centroid),
+        ClosePath()
+    ])
+end
+
+const BezierLTriangle = rotate(BezierUTriangle, pi/2)
+const BezierDTriangle = rotate(BezierUTriangle, pi)
+const BezierRTriangle = rotate(BezierUTriangle, 3pi/2)
+
+
+const BezierSquare = let
+    r = 0.95 * sqrt(pi)/2/2 # this gives a little less area as the r=0.5 circle
+    BezierPath([
+        MoveTo(Point2(r, -r)),
+        LineTo(Point2(r, r)),
+        LineTo(Point2(-r, r)),
+        LineTo(Point2(-r, -r)),
+        ClosePath()
+    ])
+end
+
+const BezierCross = let
+    cutfraction = 2/3
+    r = 0.5 # 1/(2 * sqrt(1 - cutfraction^2))
+    ri = 0.166 #r * (1 - cutfraction)
+
+    first_three = Point2[(r, ri), (ri, ri), (ri, r)]
+    all = map(0:pi/2:3pi/2) do a
+        m = Mat2f0(sin(a), cos(a), cos(a), -sin(a))
+        Ref(m) .* first_three
+    end |> x -> reduce(vcat, x)
+
+    BezierPath([
+        MoveTo(all[1]),
+        LineTo.(all[2:end])...,
+        ClosePath()
+    ])
+end
```

Comment:
Why do these have slightly different sizes, and sizes smaller than 1x1 (or radius 0.5)?

For reference, Bezier circle and square on top of 100x100px shader square:
```julia
scene = Scene()
scatter!(scene, Point2f(0), marker = Rect, color = :green, markersize = 100)
scatter!(scene, Point2f(0), marker = Makie.BezierSquare, color = :blue, markersize = 100)
scatter!(scene, Point2f(0), marker = Makie.BezierCircle, color = :red, markersize = 100)
scene
```
![Screenshot from 2022-09-10 20-17-14](https://user-images.githubusercontent.com/10947937/189496788-7f12d220-c23b-4a5b-b797-1f743970e70d.png)
and with the radii set to 0.5:
![Screenshot from 2022-09-10 20-23-10](https://user-images.githubusercontent.com/10947937/189496810-ab713327-db54-43d1-bab9-8c4d4f37a0f7.png)

Reply:
Ok so the reasoning was to create markers that have the same area, because I had assumed these would look similar. That's an important characteristic for markers because you don't really want to convey a difference in size when you use two different marker shapes with the same `markersize` setting.

I started with the circle, which I gave area = 1. That means the radius is about 0.56. Later I changed that because I wanted each shape to fit approximately within the 1x1 square. I then found out that not all shapes look like they have similar size if you correctly compute their area, probably because human visual perception introduces non-linearities when shapes differ. That's well known but I didn't exactly go on a literature search to do this (maybe I should).

So anyway, after I while I had shapes that looked good together, all approximately located in a 1x1 square. But these looked too big when mixing with character-based markers like `'X'` or `'a'`. That's because fonts at font-size X do not fill a square with size X * X with each character, they're effectively smaller. So I multiplied the whole range of markers with a factor that looked reasonably pleasing to me with both small and large letters. That is entirely subjective of course, but as markers are usually not meaningful in data space, it seemed reasonable to me to go this route. The benefit is that you can mix and match the normal markers with text/char at the same size, and won't have to adjust markersize each time. The drawback is that there's not a clean "theoretical" reason why they are of this size. I understand that making each marker fill 1x1 exactly sounds attractive theoretically, but to me it seemed to be less attractive in practice.

I find it confusing if char-based markers at size X are different than the same thing as `text` at size X, so that fixes those markers. Then the question is, should the default markers match those in size or not, I said yes they should. Hence the unusual size.

The nice thing about the new system is that it's easy for users to make their own markers, and those can conform to a base size of 1x1 or whatever, if that metric is important to the users. I can imagine it might be important when tiling markers in dataspace, but I'd argue that is sufficiently niche as to not influence the decision for a default.

Reply:
![image](https://user-images.githubusercontent.com/22495855/189618972-febb7954-ca73-4028-8dcd-294507b21207.png)

here you can see a radius 0.5 circle at the top, plus some other new default markers below, and two characters. I would find it pretty impractical if all default markers were of the size as the large circle.

Reply:
I see, that makes sense.

There's a note on characters being of nonuniform size/area but as far as I can tell we don't explain how characters or other markers are actually scaled. I think it would be good to have that information there though. I.e. with `markerspace = :pixel`
- for characters `markersize` is equivalent to `textsize/fontsize` which is the pixel height of the bounding box a character is placed into, to my understanding
- default shape markers are scaled to character markers, meaning that `markersize = 10` results in smaller side length/diameters
- custom Bezier markers use a 1:markersize scaling between units of the marker and units in the plot. I.e. a Bezier marker filling a 0..1 x 0..1 square with `markerspace = 10` will fill a 0..10 x 0..10 pixel square on the screen exactly.
- Polygons, I assume, are handled the same

Reply:
Yes it should be documented as such, I agree. And you're right, effectively the bezier coordinates are used as they are and just multiplied with the markersize, so you can get pixel-perfect output if you want.